### PR TITLE
Community campaign subpage

### PR DESCRIPTION
**Pull request by mendelB**
### What does this PR do?

This PR adds support for routing to a  'Community' Page (`/community`) on a campaign (which has a Community Page but not an `activityFeed` field).

- Adding parsing for a `customBlock` content type to the contentful Entity `parseBlocks` method (to enable handling custom blocks which exist in many an `activityFeed` and will be ported to the community pages)
- Accounting for a Community page in the `page`s field when setting the `hasCommunityPage` boolean for `CampaignPage` routing
- Adding community page specific rendering in `campaignSubPage` to render the community page with blocks. (Will be implemented for pages in general down the line)
- Rendering the `campaignSubPage` from the `/community` route on campaigns which have a community page and no activity feed
- Adding the `hideFromNavigation` field to the `Page` Entity (temporary as we want to get rid of that logic in the future)

### Any background context you want to provide?
The setup is somewhat intensely specific to the weird world we have now with balancing community pages and `activityFeed`s. Once we sunset `activityFeed` entirely and move over to pages for everything, this will be able to be cleaned up majorly!

The `isCommunity` prop is to circumvent the general methodoligy the `campaignSubPage` uses to find the appropriate page. We also don't have access to the `match` prop (and anyways don't have a `:slug`) when on the `/community` route.

### What are the relevant tickets/cards?
[#156771690](https://www.pivotaltracker.com/story/show/156771690)

### CODE REVIEW

File: app/Entities/Page.php [hunk 1]
```diff
@@ -25,6 +25,23 @@ public function parseSidebar($sidebarItems)
...
+     * Parse blocks, and reverse parsed blocks for community pages.
+     *
+     * @param  array $blocks
+     * @return array
+     */
+    public function parseBlocks($blocks)
+    {
+        $parsedBlocks = parent::parseBlocks($blocks);
+
+        if (ends_with($this->slug, 'community')) {
```
Comment (weerd) [inline app/Entities/Page.php]: Do we need to worry if someone accidentally enters a value for the slug as `some-campaign-name/community/`; ending slash?

Comment (mendelB) [reply app/Entities/Page.php]: `ends_with` wouldn't handle that. Is this a scenario we need to account for? (I guess we could use regex instead?)

File: resources/assets/components/pages/CampaignSubPage/CampaignSubPage.js [hunk 2]
```diff
@@ -15,14 +16,22 @@ import CampaignPageNavigationContainer from '../../CampaignPageNavigation/Campai
...
+    campaignEndDate,
+    isCommunity,
+    match,
+    noun,
+    pages,
+    tagline,
+    verb,
+  } = props;
+
+  const pageSlug = isCommunity ? 'community' : match.params.slug;
```
Comment (weerd) [inline resources/assets/components/pages/CampaignSubPage/CampaignSubPage.js]: Slightly confused... why wouldn't `community` be part of the `match.params.slug`? 

For `faq` and `scholarship` pages it seems to work fine and find the page in the list. If we're here to render a community page, it's already established in the `CampaignPage` that a page for community exists in the pages list.

https://github.com/DoSomething/phoenix-next/pull/892/files#diff-f22cfc1593dfc65f4beca9bf1de156b6R72

Comment (weerd) [reply resources/assets/components/pages/CampaignSubPage/CampaignSubPage.js]: Oh wait. Nevermind. Now I get it. It's not using `Route` to determine the component thus `match` is not available! Now I understand better what you meant in the description 👍

Comment (mendelB) [reply resources/assets/components/pages/CampaignSubPage/CampaignSubPage.js]: I think the issue is that this is a statically declared route `.../community` as opposed to the others where we're routing it through `.../:slug`. 

In those cases the trailing slug is set as a param under `match.params.slug`. In the case of community -- until we remove the `/community` route entirely when we do away with `activity_feed`s, we need to deal with this scenario manually.  

Also, we'd need to find the community page entry anyways, so figure we'd just use the existing subPage `find` logic.

Comment (mendelB) [reply resources/assets/components/pages/CampaignSubPage/CampaignSubPage.js]: Ah damn, you spoke! ...github 🙄

**Comment by mendelB** [comment]
@weerd We do have several other places the check for `community` slug  through JS `endsWith` (or just compare slugs in general) without the `/` protection. Do we want to add that safeguard in those places as well?
([here](https://github.com/DoSomething/phoenix-next/blob/master/resources/assets/components/pages/CampaignSubPage/CampaignSubPage.js#L24), [here](https://github.com/DoSomething/phoenix-next/pull/892/files#diff-485aa9148b8caa1bc41166a564dfdde6R19) and [here](https://github.com/DoSomething/phoenix-next/blob/master/resources/assets/helpers/index.js#L558) is what I could find offhand)

**Comment by weerd** [comment]
@mendelB I think so, yes. But separate PR and different approach since those other instances are all in JS. Maybe a similar Lodash method or just make a small helper function?

**Comment by mendelB** [comment]
👍 Will do! I'll merge this one in then.

**Pull request merged**